### PR TITLE
HDDS-6662. Provide basic hash code generator for OBSKeyPathLockStrategy

### PR DESCRIPTION
**Pull request by tanvipenumudy**
## What changes were proposed in this pull request?

This task is to provide a basic hashing module: `OMHashCodeGenerator` interface which shall be used for the `OBSKeyPathLockStrategy` class locking logic.

Note:- Now, will implement a basic String.hashCode() function, we will refine this part in a separate sub-task. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6662

## How was this patch tested?

NA

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/OMHashCodeGenerator.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package org.apache.hadoop.ozone.om.hashcodegenerator;
+
+/**
+ * Interface for hash code generation implementations.
+ */
+public interface OMHashCodeGenerator {
+
+  int getHashCode(String resourceName);
```
Comment (rakeshadr) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/OMHashCodeGenerator.java:25]: Please add javadoc

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/DefaultOMHashCodeGeneratorImpl.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.ozone.om.hashcodegenerator;
+
+/**
+ * Implementation of OMHashCodeGenerator interface.
+ */
+public class DefaultOMHashCodeGeneratorImpl implements OMHashCodeGenerator {
```
Comment (rakeshadr) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/DefaultOMHashCodeGeneratorImpl.java:23]: Please rename class `DefaultOMHashCodeGeneratorImpl`  to -> `StringOMHashCodeGeneratorImpl` to reflect the implementation. Also, please update javadoc saying that "This implementation uses String hashcode function."

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OBSKeyPathLockStrategy.java [hunk 3]
```diff
@@ -31,6 +33,9 @@
  */
 public class OBSKeyPathLockStrategy implements OzoneLockStrategy {
 
+  private OMHashCodeGenerator omHashCodeGenerator =
+      new DefaultOMHashCodeGeneratorImpl();
```
Comment (rakeshadr) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OBSKeyPathLockStrategy.java:37]: Add a TODO saying that, "TODO: need to make this pluggable and allows users to configure the preferred hashcode generation mechanism."

**Review by rakeshadr — COMMENTED** [review]
Thanks @tanvipenumudy for the contribution. Please take care the comments!

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/DefaultOMHashCodeGeneratorImpl.java (hunk 2 above)
Comment (tanvipenumudy) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/DefaultOMHashCodeGeneratorImpl.java]: I have made the changes, thank you!

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OBSKeyPathLockStrategy.java (hunk 3 above)
Comment (tanvipenumudy) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OBSKeyPathLockStrategy.java]: I have added the TODO comment, thanks!

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/OMHashCodeGenerator.java (hunk 1 above)
Comment (tanvipenumudy) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/hashcodegenerator/OMHashCodeGenerator.java:31]: I have added the Javadoc, thank you for the comment!

**Comment by adoroszlai** [comment]
@tanvipenumudy can you please resolve the conflict?

**Review by rakeshadr — APPROVED** [review]
+1 LGTM. Thanks @tanvipenumudy for the contribution.

**Pull request merged**